mlir/lib/Conversion/IndexToSPIRV/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+add_mlir_conversion_library(MLIRIndexToSPIRV
+  IndexToSPIRV.cpp
+
+  ADDITIONAL_HEADER_DIRS
+  ${MLIR_MAIN_INCLUDE_DIR}/mlir/Conversion/IndexToSPIRV
+
+  DEPENDS
+  MLIRConversionPassIncGen
+
+  LINK_COMPONENTS
+  Core
+
+  LINK_LIBS PUBLIC
+  MLIRIndexDialect
```

Comment:
Bazel is better at checking build deps. Since there's already a PR adding bazel rules, can we link with the same set of libs as bazel https://github.com/llvm/llvm-project/pull/69743/files?

Also, we can cherry-pick the bazel changes and land this in one go. (We should wake sure *not* to squash during code review so that the author of the bazel patch gets properly attributes when we hit 'merge',)